pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1132 @@ package listusers
+		  
+		  condition isTrue(param: string) {
+			param == "true"
+		  }`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKeyWithCondition("document:1", "viewer", "user:maria", "isTrue", conditionContextWithTrueParam),
+				tuple.NewTupleKeyWithCondition("document:1", "viewer", "user:jon", "isTrue", conditionContextWithFalseParam),
+				tuple.NewTupleKey("document:1", "owner", "user:will"),
+				tuple.NewTupleKey("document:1", "editor", "user:poovam"),
+			},
+			expectedUsers: []string{"user:will", "user:poovam", "user:maria"},
+		},
+	}
+	tests.runListUsersTestCases(t)
+}
+
+func TestListUsersIntersection(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	tests := ListUsersTests{
+		{
+			name:                  "intersection",
+			TemporarilySkipReason: "because incurring `panic: unexpected userset rewrite encountered`",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define required: [user]
+					define required_other: [user]
+					define viewer: required and required_other`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required", "user:will"),
+				tuple.NewTupleKey("document:1", "required_other", "user:will"),
+
+				tuple.NewTupleKey("document:1", "required", "user:jon"),
+				tuple.NewTupleKey("document:1", "required_other", "user:maria"),
+			},
+			expectedUsers: []string{"user:will"},
+		},
+		{
+			name:                  "intersection_and_ttu",
+			TemporarilySkipReason: "because incurring `panic: unexpected userset rewrite encountered`",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+		  type user
+		  type folder
+			relations
+			  define viewer: [user]
+		  type document
+			relations
+			  define required: [user]
+			  define parent: [folder]
+			  define viewer: (viewer from parent) and required`,
+
 			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required", "user:will"),
+				tuple.NewTupleKey("folder:x", "viewer", "user:will"),
 				tuple.NewTupleKey("document:1", "parent", "folder:x"),
+
+				tuple.NewTupleKey("document:1", "required", "user:maria"),
 				tuple.NewTupleKey("folder:x", "viewer", "user:jon"),
 			},
-			expectedUsers: []*openfgav1.User{
-				{
-					User: &openfgav1.User_Object{
-						Object: &openfgav1.Object{Type: "user", Id: "jon"},
+			expectedUsers: []string{"user:will"},
+		},
+	}
+	tests.runListUsersTestCases(t)
+}
+
+func TestListUsersUnion(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	tests := ListUsersTests{
```

Comment:
Just to cover the bases, let's add one more variant where we test all 3 base rewrite cases in the same union. e.g.

```
type user

type folder
  relations
    define viewer: [user]

type document
  relations
    define editor: [user]
    define viewer: [user] or editor or viewer from parent
```

The tests we have in here cover TTU and computed joined by union, but not also the directly assignable relationship included in the union.